### PR TITLE
feature: add tutorial on testing with JUnit

### PR DESCRIPTION
**Pull request by koshachy**

### CODE REVIEW

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+2. Open the `build.gradle(.kts)` file and add the following dependencies to the Gradle configuration to work with `kotlin.test` and `JUnit`:
+
+<tabs>
+
+   ```groovy
+   dependencies {
+       // Other dependencies.
+       // ...
+       testImplementation "org.jetbrains.kotlin:kotlin-test:1.3.11"
+       testImplementation "org.jetbrains.kotlin:kotlin-test-junit:1.3.11"
```
Comment (ilya-g) [inline docs/topics/jvm/jvm-test-using-junit.md:26]: Why are versions explicit here and not in the kotlin DSL example below?

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+       fun testSum() {
+           val expected = 42
+           assertEquals(expected, classForTesting.sum(40, 2))
+       }
+   }
+   ```
+
+## Run a test
+
+1. Run the test using the gutter icon. You can also run a test via the command-line interface using the `./gradlew` command.
```
Comment (ilya-g) [inline docs/topics/jvm/jvm-test-using-junit.md:95]: > You can also run a test via the command-line interface using the `./gradlew` command

I suppose using the `./gradlew check` to execute the `check` task which itself depends on tasks running project tests.
Then, to run a single test, some additional arguments may be required. So may be better to say "You can also run all project tests ..."

File: docs/topics/jvm/jvm-test-using-junit.md (hunk 1 above)
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md]: I've updated the sample to JUnit 5 and explicitly add versions to their dependencies (both Kotlin and Groovy)

File: docs/topics/jvm/jvm-test-using-junit.md (hunk 2 above)
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md]: Yeah, thanks!
I've taken your suggestion and update the command to `./gradlew test` (according to official DSL)

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+   * Define the test `testSum()` function using the [@Test annotation](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/-test/index.html).
+   * Check that the `sum()` function returns the value that is equal to the expected one using the [assertEquals()](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/-test/assert-equals.html) function.
+
+   ```kotlin
+   import kotlin.test.Test
+   import kotlin.test.assertEquals
+   
+   internal class SampleTest {
+   
+       private val classForTesting: Sample = Sample()
```
Comment (p7nov) [inline docs/topics/jvm/jvm-test-using-junit.md:101]:
```suggestion
       private val testSample: Sample = Sample()
```
 or `testInstance` seems more natural to me.

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+   ![Run the test](run-test.png)
+
+   > You can also run all project tests via the command-line interface using the `./gradlew test` command.
+   > 
+   {type="note"} 
+
+2. Check the result of the test run in the **Run** tool window:
+
+   ![Check the reult](check-the-result.png)
```
Comment (p7nov) [inline docs/topics/jvm/jvm-test-using-junit.md:123]: Typo `reult`. Also, I'd make more meaningful alt text - the image with the failed text has almost the same. Smth with `success` here and with `fail` below.

**Review by p7nov — COMMENTED** [review]
A couple of minor comments.

File: docs/topics/jvm/jvm-test-using-junit.md (hunk 2 above)
Comment (ilya-g) [reply docs/topics/jvm/jvm-test-using-junit.md]: In a JVM project, yes. In an MPP project there is no `test` task, test tasks there have a platform prefix like `jvmTest`, `jsTest` etc, and there is a unifying `check` task that runs them all.

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+   ```groovy
+   test {
+       useJUnitPlatform()
+   }
+   ```
+
+   ```kotlin
+   tasks.test {
+       useJUnit()
```
Comment (ilya-g) [inline docs/topics/jvm/jvm-test-using-junit.md:53]: Also `...JUnitPlatform`

File: docs/topics/jvm/jvm-test-using-junit.md (hunk 2 above)
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md]: We have another doc page on running test in MPP, but I'll update the command, thanks for the details!

File: docs/topics/jvm/jvm-test-using-junit.md (hunk 5 above)
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md]: Thanks!
I've updated the Groovy to JUnit 5 but forgot to update the Kotlin code block

**Review by ilya-g — APPROVED** [review]

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+   The test execution failed.
+
+## What's next
+
+Once you've finished your first test, you can:
+
+* Try to write another test using other [kotlin.test](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/) functions. For example, you could use the [`assertNotEquals()`](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-not-equals.html) function.
+* [Create your first application](jvm-spring-boot-restful.md) with Kotlin and Spring Boot.
+* Watch these video [tutorials on YouTube](https://www.youtube.com/playlist?list=PL6gx4Cwl9DGDPsneZWaOFg0H2wsundyGr), which demonstrate how to use Spring Boot with Kotlin and JUnit 5.
```
Comment (katia-energizer) [inline docs/topics/jvm/jvm-test-using-junit.md:149]: Seems strange to divide words for link text. [these video tutorials] Youtube is extra info IMO

**Review by katia-energizer — APPROVED** [review]
Overall LGTM. Minor comments

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+       // Other dependencies.
+       testImplementation(kotlin("test-junit5"))
+       testImplementation("org.junit.jupiter:junit-jupiter-api:5.6.0")
+       testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.6.0")
+   }
+   ```
+
+   </tabs>
+
+3. Check to make sure that the **Project Wizard** automatically created the `test` task in the `build.gradle(.kts)` file:
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md:41]: Yeah, agreed!

File: docs/topics/jvm/jvm-test-using-junit.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+2. Open the `build.gradle(.kts)` file and add the following dependencies to the Gradle configuration. They will allow you to work with `kotlin.test` and `JUnit`:
+
+   <tabs>
+
+   ```groovy
+   dependencies {
+       // Other dependencies.
+       testImplementation 'org.jetbrains.kotlin:kotlin-test-junit5'
+       testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md:25]: It doesn't work without it :/

[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/jvm/jvm-test-using-junit.md:25]: It doesn't work without it :/

**Pull request merged**